### PR TITLE
add as_rtf for gsBinomialExactTable

### PR DESCRIPTION
**Pull request by wangben718**
write as_rtf method for gsBinomialExactTable

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -28,7 +28,8 @@ Imports:
     tibble,
     tidyr,
     tools,
-    xtable
+    xtable,
+    r2rtf
```
Comment (nanxstats) [inline DESCRIPTION:32]: Maybe this is not too obvious but the dependencies should be listed in alphabetical orders.

File: R/as_rtf.R [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#' Print a summary table using gt
```
Comment (nanxstats) [inline R/as_rtf.R:1]: I thought this is RTF, not gt.

File: R/as_rtf.R [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#'
+#' Create RTF file created with \code{\link{as_table}} to summarize
```
Comment (nanxstats) [inline R/as_rtf.R:3]: Avoid using two consecutive "create".

File: R/as_rtf.R [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+#' an object; currently only implemented for \code{\link{gsBinomialExact}}.
+#'
+#' @param x   Object to be display on rtf.
```
Comment (nanxstats) [inline R/as_rtf.R:6]: Use only one whitespace between param name and description.

Comment (nanxstats) [reply R/as_rtf.R:6]: I would rewrite the description. "Object to be display(ed) on rtf" sounds off.

File: R/as_rtf.R [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+#' @param ... Other parameters that may be specific the object.
```
Comment (nanxstats) [inline R/as_rtf.R:7]: What does "that may be specific the object" mean?

File: R/as_rtf.R [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+#'
+#' @return NULL.
```
Comment (nanxstats) [inline R/as_rtf.R:9]: You should return the input object invisibly. See tidyverse design guide: https://design.tidyverse.org/out-invisible.html

File: R/as_rtf.R [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+#'
+#' @details
+#' Currently only implemented for \code{\link{gsBinomialExact}} objects.
+#' Creates a table to summarize an object.
+#' For \code{\link{gsBinomialExact}}, this summarized operating characteristics
+#' across a range of effect sizes.
+#'
+#' @export
+#'
+#' @examples
```
Comment (nanxstats) [inline R/as_rtf.R:21]: If you have the `@examples` section, it should have code examples and not left blank.

File: R/as_rtf.R [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+as_rtf <- function(x, ...) UseMethod("as_rtf")
+
+#' @rdname as_rtf
+#'
+#' @param title Table title.
+#' @param subtitle Table subtitle.
+#' @param theta_label Label for theta.
+#' @param bound_label Label for bounds.
+#' @param en_label Label for en.
```
Comment (nanxstats) [inline R/as_rtf.R:30]: It's unclear to me what "en" means - you should make it clear.

File: R/as_rtf.R [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+#' @param prob_decimals Number of decimal places for probability of crossing.
+#' @param en_decimals Number of decimal places for expected number of
+#'   observations when bound is crossed or when trial ends without crossing.
+#' @param rr_decimals Number of decimal places for response rates.
+#' @param rtf_path Output path for the rtf file.
```
Comment (nanxstats) [inline R/as_rtf.R:35]: I'd name this argument `file` and keep it consistent across all such functions.

```r
#' @param file File path for the output.
```

File: R/as_rtf.R [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+           subtitle = "Assumes trial evaluated sequentially after each response",
+           theta_label = "Underlying\nresponse rate",
+           bound_label = c("Futility bound", "Efficacy bound"),
+           en_label = "Expected sample sizes",
+           prob_decimals = 2,
+           en_decimals = 1,
+           rr_decimals = 0,
+           rtf_path,
+           ...) {
+    x[,2] <- sprintf(paste0("%.", prob_decimals,"f"), unlist(x[,2]))
```
Comment (nanxstats) [inline R/as_rtf.R:82]: I'm not sure if it's possible here but we should strive to access data frame or list elements by their names, not the index number.

**Review by nanxstats — CHANGES_REQUESTED** [review]
Please see my comments above.

File: R/as_rtf.R [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+#'     bound_label = c("low rate", "high rate"),
+#'     file = tempfile(fileext = ".rtf")
+#'   )
+as_rtf <- function(x, ...) UseMethod("as_rtf")
+
+#' @rdname as_rtf
+#'
+#' @param title Table title.
+#' @param theta_label Label for theta.
+#' @param response_outcome Is outcome response rate
```
Comment (LittleBeannie) [inline R/as_rtf.R:55]: Can we refine the explanation of `response_outcome` as `Logical values indicating if the outcome is response rate (TRUE) or failure rate (FALSE).

**Review by LittleBeannie — APPROVED** [review]

File: R/as_rtf.R (hunk 6 above)
Comment (nanxstats) [reply R/as_rtf.R]: Note that the returned object should be the **original input**, not the modified object.

**Comment by nanxstats** [comment]
I updated the code to remove the check note on `title` \usage more than 90 characters wide and improved the general code and documentation style in both `as_rtf()` and `as_gt()`.

Most importantly, I now put `...` right after required arguments following the [tidyverse design guide](https://design.tidyverse.org/dots-after-required.html) to avoid potential coding pitfalls for users.

I just have some final minor questions regarding the RTF output style and consistency generated by the example code:

1. In the first output, the second "Probability of crossing" column has NA as the sub-column name.
2. In the second output, the two columns are named "low rate" and "high rate" - should they use sentence case as the other columns?
3. In both outputs, should column names use sentence case or title case?
4. In both outputs, should table title use sentence case or title case?

**Comment by nanxstats** [comment]
Thanks for improving the details in the output tables. It turns out for a length-1 input, `ifelse()` (as designed) returns only the first value, so using `if else` instead is the proper way.

**Pull request merged**